src/main/java/com/commercetools/sync/categories/utils/CategoryUpdateActionUtils.java

```diff
@@ -0,0 +1,211 @@
+     * {@link Optional} is returned.
+     * <p>
+     * Note: If the parent {@link Reference<Category>} of the new {@link CategoryDraft} is null, an empty
+     * {@link Optional} is returned with no update actions and a custom callback function, if set on the
+     * supplied {@link CategorySyncOptions}, is called.
+     *
+     * @param oldCategory the category which should be updated.
+     * @param newCategory the category draft where we get the new parent.
+     * @param syncOptions     the sync syncOptions with which a custom callback function is called in case the parent is null.
+     * @return A filled optional with the update action or an empty optional if the parent references are identical.
+     */
+    @Nonnull
+    public static Optional<UpdateAction<Category>> buildChangeParentUpdateAction(
+            @Nonnull final Category oldCategory,
+            @Nonnull final CategoryDraft newCategory,
+            @Nonnull final CategorySyncOptions syncOptions) {
+        final Reference<Category> newCategoryParentReference = newCategory.getParent();
+        if (newCategoryParentReference == null) {
+            syncOptions.callUpdateActionWarningCallBack(
+                    format(CATEGORY_CHANGE_PARENT_EMPTY_PARENT, oldCategory.getId()));
+            return Optional.empty();
+        }
+        return buildUpdateAction(oldCategory.getParent(),
+                newCategoryParentReference,
+                ChangeParent.of(newCategoryParentReference));
+    }
+
+    /**
+     * Compares the orderHint values of a {@link Category} and a {@link CategoryDraft} and returns an
+     * {@link UpdateAction<Category>} as a result in an {@link Optional}. If no update action is needed, for example in
+     * case where both the {@link Category} and the {@link CategoryDraft} have the same orderHint values, an empty
+     * {@link Optional} is returned.
+     * <p>
+     * Note: If the orderHint of the new {@link CategoryDraft} is null, an empty {@link Optional} is returned with
+     * no update actions and a custom callback function, if set on the supplied {@link CategorySyncOptions}, is called.
+     *
+     * @param oldCategory the category which should be updated.
+     * @param newCategory the category draft where we get the new orderHint.
+     * @param syncOptions     the sync syncOptions with which a custom callback function is called in case the orderHint is null.
+     * @return A filled optional with the update action or an empty optional if the orderHint values are identical.
+     */
+    @Nonnull
+    public static Optional<UpdateAction<Category>> buildChangeOrderHintUpdateAction(
+            @Nonnull final Category oldCategory,
+            @Nonnull final CategoryDraft newCategory,
+            @Nonnull final CategorySyncOptions syncOptions) {
+        final String newCategoryOrderHint = newCategory.getOrderHint();
+        if (newCategoryOrderHint == null) {
+            syncOptions.callUpdateActionWarningCallBack(
```

Comment:
because the new fields for update actions for category description, orderHint and parent are required fields by our API; as opposed to the rest of the update actions for categories: https://github.com/commercetools/commercetools-sync-java/issues/8

I added more unit tests that show that other fields can still build update actions with null fields: b90f62a